infrastructure/src/main/java/org/corfudb/infrastructure/SequencerServer.java

```diff
@@ -296,8 +294,7 @@ private TxResolutionResponse txnCanCommit(TxResolutionInfo txInfo) {
             // for each key pair, check for conflict; if not present, check against the wildcard
             for (byte[] conflictParam : conflictParamSet) {
 
-                Long keyAddress = cache.get(new ConflictTxStream(conflictStream.getKey(),
-                        conflictParam, Address.NON_ADDRESS));
+                Long keyAddress = cache.get(new ConflictTxStream(conflictStream.getKey(), conflictParam));
```

Comment:
I think there is a correctness issue here. An abort happens when the intersection of two conflict windows is not empty. In this case, you can have two transactions intersecting at one version.
For example:

```
Version:  1     2     3     4     5     6     7
Txn1:                 S     E
Txn2:                       S     E
```

The start (S) and (E) of Txn1 intersects with Txn2 at version 4, but the transaction would not abort because of the following condition

```java

if (keyAddress > txSnapshotTimestamp.getSequence()) {

```

So shouldn't it be `>=` instead of `>`  ?

Reply:
My example is wrong, you can ignore this comment. The transactions don't intersect because Txn1's window is [3, 4) and Txn2's window is [4, 5). Essentially, Txn2 starts its transaction with Txn1 commit address as its snapshot.